helm-charts/splunk-otel-collector/templates/operator/instrumentation.yaml

```diff
@@ -19,29 +19,30 @@ spec:
     - tracecontext
     - baggage
     - b3
-  {{- if .Values.operator.instrumentation.spec.sampler }}
-  {{ include .Values.operator.instrumentation.spec.sampler }}
+  {{- with .Values.operator.instrumentation.spec.sampler }}
+  sampler:
+  {{- toYaml . | nindent 4 }}
```

Comment:
This seems like a fix of another issue? If so, please submit a separate PR. Let's try to keep the changeset serving one urpose

Reply:
This is needed to fix the regression. I'll fix only the regressions and move the new example out to a new PR.

Reply:
I understand that this is a regression, but it's another regression that is unrelated to the reported issue which is meant to be fixed in this PR, right? The reported issue is about `instrumentation.spec.env` not `instrumentation.spec.sampler`

Reply:
I've updated the description of this PR to outline the changes more accurately. I'll try to use GH issues more appropriately in the future. For at least this time since this change-set is small I would prefer not to split this PR out more.

Reply:
It's not about issues specifically. We just need to make sure each PR addresses a particular problem/improvement outlined in the description or the linked issue. It makes the changeset clean for the git history and makes the review process much less confusing.